shared/src/main/scala/scorex/crypto/hash/hash.scala

```diff
@@ -1,24 +1,39 @@
 package scorex.crypto
 
-import supertagged.TaggedType
+import scorex.crypto.authds.ADKey
+import scorex.crypto.hash.Sha256.Message
 
 package object hash {
-
-  trait BaseDigest extends TaggedType[Array[Byte]]
-
-  type Digest = BaseDigest#Type
-
-  object Digest32 extends BaseDigest
-
-  type Digest32 = Digest32.Type
-
-  object Digest64 extends BaseDigest
-
-  type Digest64 = Digest64.Type
-
-  object NonStandardDigest extends BaseDigest
-
-  type NonStandardDigest = NonStandardDigest.Type
+  implicit def digest32ToArray(dig: Digest32): Array[Byte] = dig.value
+  implicit def digest32ToMessage(dig: Digest32): Message = dig.value
+  implicit def digest64ToArray(dig: Digest64): Array[Byte] = dig.value
+  implicit def digestToArray(dig: Digest): Array[Byte] = dig.value
+  trait Digest extends Any {
+    val value: Array[Byte]
+    def toList: List[Byte] = value.toList
+    def ++(c: Array[Byte]): Array[Byte] = value ++ c
+    def sameElements(c: Array[Byte]): Boolean = value.sameElements(c)
+    def isEmpty: Boolean = value.isEmpty
+    def :+(c: Byte): Array[Byte] = value :+ c
+    def take(num: Int): Array[Byte] = value.take(num)
```

Comment:
this methods doesn't look good, each digest has fixed length, so why all these operations?